src/schema/schema_create.c

```diff
@@ -403,6 +403,14 @@ __create_index(WT_SESSION_IMPL *session,
 		WT_RET_MSG(session, EINVAL, "Invalid index name, "
 		    "should be <table name>:<index name>: %s", name);
 
+	/* Check if the file already exists. */
+	if ((ret = __wt_metadata_search(session, name, &origconf)) !=
+	    WT_NOTFOUND) {
+		if (exclusive)
+			WT_TRET(EEXIST);
+		exists = true;
+	}
```

Comment:
I think we need a `WT_ERR(ret);` here -- otherwise we're assuming `origconf` is valid below...